goal_src/levels/rolling/rolling-lightning-mole.gc

```diff
@@ -5,3 +5,1518 @@
+      (let ((a1-0 (new-stack-vector0)))
+        (clmf-input gp-0 a1-0 1)
+        )
+      (set! (-> self run-blend-interp) (-> gp-0 y))
+      )
+    (set! (-> self speed-adjust) 1.0)
+    (none)
+    )
+  :code
+  lightning-mole-run-code
+  :post
+  (the-as (function none :behavior lightning-mole) ja-post)
+  )
+
+(defstate lightning-mole-debug-run (lightning-mole)
+  :enter
+  (behavior ()
+    (set! (-> self speed-adjust) 1.0)
+    (set-vector! (-> self debug-vector) 0.0 0.0 1.0 1.0)
+    (set! (-> self saved-travel quad) (-> self debug-vector quad))
+    ((-> (method-of-type nav-enemy nav-enemy-chase) enter))
+    (none)
+    )
+  :trans
+  (behavior ()
+    (when (cpad-pressed? 1 r3)
+      (logclear! (-> *cpad-list* cpads 1 button0-abs 0) (pad-buttons r3))
+      (logclear! (-> *cpad-list* cpads 1 button0-rel 0) (pad-buttons r3))
+      (go lightning-mole-debug-run)
+      )
+    (none)
+    )
+  :code
+  lightning-mole-run-code
+  :post
+  (behavior ()
+    (let ((gp-0 (new 'stack-no-clear 'vector)))
+      (let ((s4-0 (new-stack-vector0))
+            (a1-0 (new-stack-vector0))
+            (s5-0 (new-stack-matrix0))
+            )
+        (clmf-input s4-0 a1-0 1)
+        (vector-float*! s4-0 s4-0 1820.4445)
+        (matrix-axis-angle! s5-0 (-> *camera* local-down) (-> s4-0 y))
+        (vector-matrix*! (-> self debug-vector) (-> self debug-vector) s5-0)
+        )
+      (vector-normalize! (-> self debug-vector) (-> self flee-info deflection-max-dist))
+      (camera-line-rel
+        (-> self collide-info trans)
+        (-> self debug-vector)
+        (new 'static 'vector4w :x #x80 :y #x80 :w #x80)
+        )
+      (set! (-> gp-0 quad) (-> self debug-vector quad))
+      (set! (-> self target-speed) (-> self nav-info run-travel-speed))
+      (format *stdcon* "tgt-speed ~M defmd ~M~%" (-> self target-speed) (-> self flee-info deflection-max-dist))
+      (if (>= (- (-> *display* base-frame-counter) (the-as int (-> self last-reflection-time)))
+              (-> self flee-info reflection-time)
+              )
+          (vector-normalize-copy! (-> self desired-travel) gp-0 (-> self flee-info deflection-max-dist))
+          )
+      )
+    (if (fleeing-nav-enemy-clip-travel self (-> self desired-travel))
+        (set! (-> self last-reflection-time) (the-as uint (-> *display* base-frame-counter)))
+        )
+    (set! (-> self debug-vector quad) (-> self desired-travel quad))
+    (let ((t9-8 fleeing-nav-enemy-adjust-travel)
+          (a0-15 self)
+          )
+      (-> self desired-travel)
+      (t9-8 a0-15)
+      )
+    (fleeing-nav-enemy-chase-post-func)
+    (camera-line-rel
+      (-> self collide-info trans)
+      (-> self desired-travel)
+      (new 'static 'vector4w :x #x80 :w #x80)
+      )
+    (camera-line-rel (-> self collide-info trans) (-> self nav travel) (new 'static 'vector4w :z #x80 :w #x80))
+    (camera-line (-> self collide-info trans) (-> self nav target-pos) (new 'static 'vector4w :z #x80 :w #x80))
```

Comment:
probably a project for later, but I think all these `camera-line` things are some weird rgba type.